src/status_im/contexts/communities/actions/accounts_selection/view.cljs

```diff
@@ -32,11 +34,19 @@
      :community-name  community-name
      :container-style {:margin-top 8}}]])
 
-(defn view
+(defn f-view-internal
   []
   (let [{id :community-id}          (rf/sub [:get-screen-params])
         {:keys [name color images]} (rf/sub [:communities/community id])
-        accounts                    (rf/sub [:wallet/accounts-with-customization-color])]
+        accounts                    (rf/sub [:wallet/accounts-with-customization-color])
+        addresses-for-permissions   (rf/sub [:communities/addresses-for-permissions])
+        selected-accounts           (filter #(contains? addresses-for-permissions
+                                                        (:address %))
+                                            accounts)]
+    (rn/use-effect (fn []
```

Comment:
@ilmotta, If I'm not mistaken, with form-2 I will also have to move `accounts` subscription to the outer function.

```cljs
(defn view
  []
  (let [accounts (rf/sub [:wallet/accounts-with-customization-color])]
    (rf/dispatch [:communities/set-addresses-for-permissions
                  (set (map :address accounts))])
    (fn []
      (let [{id :community-id}          (rf/sub [:get-screen-params])
            {:keys [name color images]} (rf/sub [:communities/community id])
```

This looks a bit more complicated to me compared to the use-effect approach.

Reply:
Fair enough :) It would be more about the performance advantage because functional components in Reagent are slower. Thanks, it's fine to keep the code as is.